test/host_tests/sal_udpserver.py

```diff
@@ -0,0 +1,175 @@
+                print "HOST: %d bytes sendto() client %s" % (len, str(self.client_address))
+                            
+            if 'shutdown' in data:
+                self.server._shutdown_request = True
+            else:
+                tx_bytes = socket.sendto(data, self.client_address)
+                print("HOST: Sent %d bytes of data" % tx_bytes)
+            
+        except Exception as e:
+            print("HOST: detected unexpected exception: %s" % str(e))
+                
+
+class SalUdpServerTest(BaseHostTest):
+    """
+    mbed greentea framework host test script for udp_echo_client server 
+    side functionality. The test does the following:
+      - creates a UDP Server for echo back to sender any packets received.
+      - communicates the udp server {ipaddr, port} to the DUT via serial
+        so the DUT can send packets to the udp server.
+      - The DUT will send udp packets of various lengths and the UDP server 
+        will echo them back again.
+      - When finished, the DUT will send a shutdown command to the UDP 
+        server causing the udp server thread to terminate, and this 
+        function to return.    
+    """
+    
+    name = 'sal_udpserver'
+
+    def send_server_ip_port(self, selftest, ip_address, port_no):
+        """send the udp server {ipaddr, port} to target via serial console."""
+
+        self.watchdog = 0
+        
+        # Read 3 lines which are sent from client
+        print "HOST: About to read 3 lines from target before sending UDP Server {ipaddr, port} tuple." 
+        for i in range(0, 3):
+            c = selftest.mbed.serial_readline()
+            if c is None:
+                selftest.print_result(self.RESULT_IO_SERIAL)
+                return
+            print "MBED: " + c.strip()
+
+        print "HOST: Sending server IP Address to target..."
+        connection_str = ip_address + ":" + str(port_no) + "\n"
+        selftest.mbed.serial_write(connection_str)
+
+        # mbed greentea framwork support the following for outputing serial console emitted 
+        # from the target, but it doesnt work relibly for me.
+        # selftest.dump_serial()
+        # selftest.dump_serial_end()
+        
+        # until its fixed, the following emits the serial console trace
+        while True:
+            c = selftest.mbed.serial_readline()
+            if c is None:
+                selftest.print_result(self.RESULT_IO_SERIAL)
+                return
+            print "MBED: " + c.strip()
+            
+            # look for the end tag in serial output denoting the test has 
+            # finished, and this can return.
+            if c.strip() == "{{end}}":
+                print "HOST: Terminating Test"
+                break
+        
+            # null lines are periodically generated, which can be used to trigger the watchdog
+            elif c.strip() == "":
+                self.watchdog += 1
+                if self.watchdog > 10:
```

Comment:
The previous watchdog uses 20.0, while this one uses 10. Is there a reason for this difference? Should they be the same? Should they be configurable?